src/qmapshack/helpers/CWptIconManager.cpp

```diff
@@ -314,5 +314,21 @@ QMenu* CWptIconManager::getWptIconMenu(const QString& title, QObject* obj, const
     }
   }
 
+  // Workaround for https://bugs.kde.org/show_bug.cgi?id=496930.
+  // Apply a Qt default style in this case
+  QRect screenGeometry = parent->windowHandle()->screen()->geometry();
+  QSize menuPreferredSize = menu->sizeHint();
+
+  if (menuPreferredSize.height() > screenGeometry.height()) {
+    qDebug() << "Found a misbehaving style: " << QApplication::style()->objectName();
+
+    QStyle* style = QStyleFactory::create("fusion");
+
+    if (style != nullptr){
+      qDebug() << "Applying a default style for widget QMenu";
+      menu->setStyle(style);
```

Comment:
setStyle does not transfer ownership to QMenu.  In other words `style` is created, used by the QMenu object but never destroyed. This is a memory leak.

https://doc.qt.io/qt-6/qwidget.html#setStyle

I still have a bad feeling about this hack. The documentation clearly states to avoid it for normal use. And if I have learned something from Qt documentation it's to listen to their warnings.

Reply:
Would it be sufficient to add `delete style;` directly after `menu->setStyle(style)` to prevent the memory leak? I tested this and it works. If this is sufficient i will change also the other things you mentioned.

Anyway, if you have bad feelings we (i) should find another solution. The warning is surely there on purpose… but i guess it just states "Use a konsistent style for your application!".

Reply:
The problem with the style instance is that is it used by the menu instance. And this is passed as return value for further use. If you delete the style object, the pointer stored in the menu object points to invalid memory. Linux is pretty sloppy if it comes to that situation. As long as the freed memory is not corrupted by another allocation it works. So this is a ticking time bomb. Windows is more strict on that and will crash. On Linux you need to compile with ASAN to catch such stuff.

"Use a konsistent style for your application!" or "This is just some internal hack and we change that for any reason. Use at your own risk".

I still wonder why this is broken on some systems. Do you use the Qt version shipped with the distribution or the one provided by Qt via their online install tool?

Reply:
Thanks for the detailed explanation.
After reading about this here and there i found `setParent()` [see here the last chapter](https://doc.qt.io/qt-6/qproxystyle.html#details) which should solve this, imho:

~~~~
    QStyle* style = QStyleFactory::create("fusion");
    style->setParent(menu);

    if (style != nullptr){
      menu->setStyle(style);
    }
~~~~

Using a proxy style might be another solution, but it is not guaranteed that it works. And digging through the QStyle documentation produces a lot of question marks in my brain. I even do not know how find out if the menu has e.g. a scroll option nor i have an idea how to add a scroll option. But this would be the most sufficient solution imho: Check if the menu have multiple columns (or a scroll option), and adjust it if not.

> I still wonder why this is broken on some systems. Do you use the Qt version shipped with the distribution or the one provided by Qt via their online install tool?

I used the version from archlinux. As far i can see they apply only two compiler related [patches](https://gitlab.archlinux.org/archlinux/packaging/packages/qt6-base).  Maybe i will try to use the online installation tool, but i fear this will mess up my installed packages, or the package manager database.